react/src/components/elements/upcoming-bill/UpcomingBill.tsx

```diff
@@ -53,31 +53,29 @@ export const UpcomingBill = forwardRef<
 
   const { data, settings, stripe } = useEmbed();
 
-  const { latestInvoice } = useMemo(() => {
+  const { upcomingInvoice } = useMemo(() => {
     return {
-      latestInvoice: {
-        ...(data.subscription?.latestInvoice?.amountDue && {
-          amountDue: data.subscription.latestInvoice.amountDue,
+      upcomingInvoice: {
+        ...(data.upcomingInvoice?.amountDue && {
+          amountDue: data.upcomingInvoice.amountDue,
         }),
         ...(data.subscription?.interval && {
           interval: data.subscription.interval,
         }),
-        ...(data.subscription?.latestInvoice?.dueDate && {
-          dueDate: toPrettyDate(
-            new Date(data.subscription.latestInvoice.dueDate),
-          ),
+        ...(data.upcomingInvoice?.dueDate && {
+          dueDate: toPrettyDate(new Date(data.upcomingInvoice.dueDate)),
         }),
       },
     };
   }, [data.subscription]);
```

Comment:
Can we add `data.upcomingInvoice` as another dependency here?

Reply:
good catch, done